### PR TITLE
chore(ts-paths): add ts jest support

### PR DESCRIPTION
**Pull request by AutanaSoft**
## Proposed changes

added the use of paths in the tsconfig.json configuration to allow shorter paths in the import.

added the ts-jest package so that jest can resolve paths correctly.

added the -r tsconfig-paths/register argument in the start:dev script as it is necessary for swc to recognize paths.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by AlbertHernandez — COMMENTED** [review]
I really like this change, I think is going to be super useful for our projects so thank you really much for this contribution @AutanaSoft ❤️.

I have just added couple of questions / thoughts I was having, nothing blocking so let's see if we can finish discussion about just defining in one place the paths and once we close that point we can go ahead with this change and have it in the template 🚀 .

Once we adopt it here I will make the same change in the other templates since I see the potential of this 🤩

**Comment by AlbertHernandez** [comment]
@AutanaSoft I can see some github actions failing, I guess is because the contribution is comming from a fork without permission in the repo so don't worry. Please, resolve the `package-lock.json` conflicts and we can merge it

File: jest.config.ts [hunk 1]
```diff
@@ -11,6 +12,10 @@ const config: Config = {
   setupFiles: ["<rootDir>.jest/set-env-vars.ts"],
   clearMocks: true,
   coverageReporters: ["json", "lcov", "text"],
+  modulePaths: ["."],
+  moduleNameMapper: pathsToModuleNameMapper({
```
[the remark this answers is not in the document]
Comment (AlbertHernandez) [reply jest.config.ts:16]: After investigating and playing a bit with this, I have found a good solution were we can import the tsconfig.json file and use it here, for doing this we need:

1. Modify the tsconfig.json for adding the `"resolveJsonModule": true` in the compiler options.
2. Modify this jest.config.ts file with the following config:

```ts
import type { JestConfigWithTsJest } from "ts-jest";
import { pathsToModuleNameMapper } from "ts-jest";

import { compilerOptions } from "./tsconfig.json";

const config: JestConfigWithTsJest = {
  ...,
  modulePaths: [compilerOptions.baseUrl],
  moduleNameMapper: pathsToModuleNameMapper(compilerOptions.paths),
};

export default config;
```

@AutanaSoft could you try this approach??

[the remark this answers is not in the document]
Comment (AutanaSoft) [reply jest.config.ts]: Hi, I have tried the method and it works.

File: package.json [hunk 2]
```diff
@@ -73,6 +73,7 @@
     "rimraf": "^5.0.5",
     "source-map-support": "^0.5.21",
     "supertest": "^6.3.4",
+    "ts-jest": "^29.1.1",
```
[the remark this answers is not in the document]
Comment (AutanaSoft) [reply package.json:76]: ts-jest is just for jest to understand the paths that are configured in tsconfig.json.

to make SWC work with the paths just add -r tsconfig-paths/register in the start:dev script.

[the remark this answers is not in the document]
Comment (AlbertHernandez) [reply package.json:76]: Yep, I would love to see this path integration in the `swc/jest`, but maybe is not threre yet, but don't worry, we can go with this version and if in the future is integrated in the other tool, we can change it

File: jest.config.ts [hunk 3]
```diff
@@ -1,6 +1,10 @@
-import type { Config } from "jest";
+import type { JestConfigWithTsJest } from "ts-jest";
+import { pathsToModuleNameMapper } from "ts-jest";
 
-const config: Config = {
+import { compilerOptions } from "./tsconfig.json";
+
+const config: JestConfigWithTsJest = {
+  preset: "ts-jest",
```
Comment (AlbertHernandez) [inline jest.config.ts:7]: **question (non-blocking):** not sure if we really need this preset, I integrated also ts-paths in [this](https://github.com/AlbertHernandez/express-typescript-service-template/blob/22462ffce8ca4e68229e79862b870d4537db517a/jest.config.ts#L8) other template and this was not needed

**Review by AlbertHernandez — APPROVED** [review]

**Comment by AlbertHernandez** [comment]
I am going to merge suggested changes, I will iterate a bit more this solution in the next iteration, again, @AutanaSoft thank you for this proposal, I think it looks amazing ❤️ 🚀

**Pull request merged**